File: server/application/src/main/kotlin/fr/devlille/partners/connect/events/domain/EventStats.kt
package fr.devlille.partners.connect.events.domain

import fr.devlille.partners.connect.partnership.domain.PartnershipItem
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class EventStats(
    val partners: List<PartnerStats>,
)

@Serializable
data class PartnerStats(
    val partnership: PartnershipItem,
    @SerialName("job_offers")
    val jobOffers: JobOfferStats,
    val activities: Int,
    val qanda: QandaStats,
    val tickets: Int,
    @SerialName("social_links")
    val socialLinks: Int,
    @SerialName("communication_plan")
    val communicationPlan: Int,
    val speakers: Int,
)

@Serializable
data class JobOfferStats(
    val total: Int,
    val validated: Int,
)

@Serializable
data class QandaStats(
    val questions: Int,
    val answers: Int,
)

File: front/utils/eventStats.ts
import { customFetch } from '../custom-instance';
import type { PartnershipItemSchema } from './api';

export interface JobOfferStatsSchema {
  total: number;
  validated: number;
}

export interface QandaStatsSchema {
  questions: number;
  answers: number;
}

export interface PartnerStatsSchema {
  partnership: PartnershipItemSchema;
  job_offers: JobOfferStatsSchema;
  activities: number;
  qanda: QandaStatsSchema;
  tickets: number;
  social_links: number;
  communication_plan: number;
  speakers: number;
}

export interface EventStatsSchema {
  partners: PartnerStatsSchema[];
}

/**
 * Per-partner engagement stats for an event.
 * @summary Get event stats for organisers
 */
export const getEventStats = (orgSlug: string, eventSlug: string) => {
  return customFetch<EventStatsSchema>({
    url: `/orgs/${orgSlug}/events/${eventSlug}/stats`,
    method: 'GET',
  });
};
